forge-gui/res/cardsfolder/upcoming/unyielding_gatekeeper.txt

```diff
@@ -0,0 +1,12 @@
+Name:Unyielding Gatekeeper
+ManaCost:1 W
+Types:Creature Elephant Cleric
+PT:3/2
+K:Disguise:1 W
+T:Mode$ TurnFaceUp | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ When CARDNAME is turned face up, exile another target nonland permanent. If you controlled it, return it to the battlefield tapped. Otherwise, its controller creates a 2/2 white and blue Detective creature token.
+SVar:TrigExile:DB$ ChangeZone | IsCurse$ True | ValidTgts$ Permanent.nonLand+Other | TgtPrompt$ Choose another target nonland permanent | Origin$ Battlefield | Destination$ Exile | RememberLKI$ True | SubAbility$ DBReturn
+SVar:DBReturn:DB$ ChangeZone | Defined$ Remembered | Origin$ Exile | ConditionDefined$ Remembered | ConditionPresent$ Card.YouCtrl | Destination$ Battlefield | SubAbility$ DBToken 
+SVar:DBToken:DB$ Token | TokenScript$ wu_2_2_detective | TokenOwner$ RememberedController | ConditionDefined$ Remembered | ConditionPresent$ Card.YouCtrl | ConditionCompare$ EQ0 | SubAbility$ DBCleanup
```

Comment:
what happens if you exile something you control but didn't own?

would it still try to create the token? (but it shouldn't)

Reply:
didn't get the token in my test

Reply:
make sure you also retain control if you didn't own it

Reply:
@tool4ever if you control, but don't own it,
then it just returns under its owners control

Reply:
@Hanmac 
I would agree normally, but they already ruled otherwise :/

> If the controller of Unyielding Gatekeeper's last ability controlled the target permanent at the time the ability began to resolve, the permanent will return to the battlefield tapped under that player's control, regardless of who its owner is.

Reply:
Any pointers to cards I should look at ?

Reply:
Just use GainControl$...